src/components/mongo.ts

```diff
@@ -0,0 +1,409 @@
+      `${this.name}-ecs-task-exec-role`,
+      {
+        namePrefix: `${this.name}-ecs-task-exec-role-`,
+        assumeRolePolicy,
+        managedPolicyArns: [
+          'arn:aws:iam::aws:policy/CloudWatchFullAccess',
+          'arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess',
+        ],
+        inlinePolicies: [
+          secretManagerSecretsInlinePolicy,
+          ...argsWithDefaults.taskExecutionRoleInlinePolicies,
+        ],
+        tags: commonTags,
+      },
+      { parent: this },
+    );
+
+    const execCmdInlinePolicy = {
+      name: `${this.name}-ecs-exec`,
+      policy: JSON.stringify({
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Sid: 'AllowContainerToCreateECSExecSSMChannel',
+            Effect: 'Allow',
+            Action: [
+              'ssmmessages:CreateControlChannel',
+              'ssmmessages:CreateDataChannel',
+              'ssmmessages:OpenControlChannel',
+              'ssmmessages:OpenDataChannel',
+            ],
+            Resource: '*',
+          },
+        ],
+      }),
+    };
+
+    const taskRole = new aws.iam.Role(
+      `${this.name}-ecs-task-role`,
+      {
+        namePrefix: `${this.name}-ecs-task-role-`,
+        assumeRolePolicy,
+        inlinePolicies: [
+          execCmdInlinePolicy,
+          ...argsWithDefaults.taskRoleInlinePolicies,
+        ],
+        tags: commonTags,
+      },
+      { parent: this },
+    );
+
+    const parsedSize = pulumi.all([argsWithDefaults.size]).apply(([size]) => {
+      const mapCapabilities = ({ cpu, memory }: CustomSize) => ({
+        cpu: String(cpu),
+        memory: String(memory),
+      });
+      if (typeof size === 'string') {
+        return mapCapabilities(PredefinedSize[size]);
+      }
+      if (typeof size === 'object') {
+        return mapCapabilities(size);
+      }
+      throw Error('Incorrect EcsService size argument');
+    });
+
+    const taskDefinition = new aws.ecs.TaskDefinition(
+      `${this.name}-task-definition`,
+      {
+        family: `${this.name}-task-definition-${stack}`,
+        networkMode: 'awsvpc',
+        executionRoleArn: taskExecutionRole.arn,
+        taskRoleArn: taskRole.arn,
+        cpu: parsedSize.cpu,
+        memory: parsedSize.memory,
+        requiresCompatibilities: ['FARGATE'],
+        containerDefinitions: pulumi
+          .all([
+            this.name,
+            argsWithDefaults.image,
+            argsWithDefaults.port,
+            argsWithDefaults.environment,
+            argsWithDefaults.secrets,
+            this.logGroup.name,
+            awsRegion,
+          ])
+          .apply(
+            ([
+              containerName,
+              image,
+              port,
+              environment,
+              secrets,
+              logGroup,
+              region,
+            ]) => {
+              return JSON.stringify([
+                {
+                  readonlyRootFilesystem: false,
```

Comment:
After disabling that, we can set this to true.

Reply:
Be sure to test ssm connect after setting this to true because I don't know will aws be able to install ssm agent in the container after we change this

Reply:
Looks like this is necessary after all. Following error occurs after setting the root system to read only.

<img width="1569" alt="Screenshot 2023-11-09 at 00 14 25" src="https://github.com/ExtensionEngine/infra-code-blocks/assets/139864784/6c4ebe31-a42d-420c-bbb0-a0655c886cd7">